docs/en/testing/supplemental.md

```diff
@@ -1,378 +1,454 @@
 
-```ts
-v('div', {
-	'classes': [
-		'root',
-(A)		'other'
-(E)		'another'
-	],
-	'onclick': 'function'
-}, [
-	v('span', {
-		'classes': 'span',
-		'id': 'random-id',
-		'key': 'label',
-		'onclick': 'function',
-		'style': 'width: 100px'
-	}, [
-		'hello 0'
-	])
-	w(ChildWidget, {
-		'id': 'random-id',
-		'key': 'widget'
-	})
-	w('registry-item', {
-		'id': true,
-		'key': 'registry'
-	})
-])
+```tsx
+.replace(wrapped: Wrapped, node: DNode);
 ```
 
-## `harness.trigger`
+#### `assertionTemplate.remove()`
 
-`harness.trigger()` calls a function with the `name` on the node targeted by the `selector`.
+Returns a new assertion template removing the target wrapped node completely.
 
-```ts
-interface FunctionalSelector {
-	(node: VNode | WNode): undefined | Function;
-}
-
-trigger(selector: string, functionSelector: string | FunctionalSelector, ...args: any[]): any;
+```tsx
+.remove(wrapped: Wrapped);
 ```
 
--   `selector`: The selector query to find the node to target
--   `functionSelector`: Either the name of the function to call from found node's properties or a functional selector that returns a function from a nodes properties.
--   `args`: The arguments to call the located function with
+#### `assertionTemplate.setProperty()`
 
-Returns the result of the function triggered if one is returned.
+Returns a new assertion template with the updated property for the target wrapped node.
 
-Example Usage(s):
-
-```ts
-// calls the `onclick` function on the first node with a key of `foo`
-h.trigger('@foo', 'onclick');
+```tsx
+.setProperty<T, K extends keyof T['properties']>(
+  wrapped: Wrapped<T>,
+  property: K,
+  value: T['properties'][K]
 ```
 
-```ts
-// calls the `customFunction` function on the first node with a key of `bar` with an argument of `100`
-// and receives the result of the triggered function
-const result = h.trigger('@bar', 'customFunction', 100);
+#### `assertionTemplate.setProperties()`
+
+Returns a new assertion template with the updated properties for the target wrapped node.
+
+```tsx
+.setProperties<T>(
+  wrapped: Wrapped<T>,
+  value: T['properties'] | PropertiesComparatorFunction<T['properties']>
+): AssertionTemplateResult;
 ```
 
-A `functionalSelector` can be used return a function that is nested in a widget's properties. The function will be triggered, in the same way that using a plain string `functionSelector`.
+A function can be set in place of the properties object to return the expected properties based of the actual properties.
 
-### Trigger example
+## Triggering Properties
 
-Given the following VDOM structure:
+In addition to asserting the output from a widget, widget behavior can be tested by using the `renderer.property()` function. The `property()` function takes a [wrapped test node]() and the key of a property to call before the next call to `expect()`.
```

Comment:
It can take additional arguments to pass to the function still right? Maybe worth mentioning

Reply:
Updated